server/src/utils/fight/fightMethods.ts

```diff
@@ -945,26 +957,31 @@ const activateSuper = (
             a: StepType.Leave,
             f: pet.index,
           });
-
           // Remove pet from fight
           fightData.fighters = fightData.fighters
             .filter((f) => f.index !== pet.index);
+        } else {
+          // Set remaining pets to play just after
+          pet.initiative = fighter.initiative - 0.01;
+          unafraidPetIndexes.push(pet.index);
         }
       }
 
-      // Abort if no pet feared
-      if (fearSteps.length === 0) return false;
+      // Increase opponents initiative
+      getOpponents({ fightData, fighter, bruteOnly: true })
+        .forEach((opponent) => opponent.initiative += 0.2 );
+
+      // Add fear steps
+      fightData.steps = fightData.steps.concat(fearSteps);
 
       // Add skill activation step
       fightData.steps.push({
         a: StepType.SkillActivate,
         b: fighter.index,
         s: SkillByName[skill.name],
+        p: unafraidPetIndexes,
```

Comment:
p here should never be defined by default, as it bloats the generated steps for nothing most of the time.
The step has to be stored in a const without the p, and the p added in an if before adding the step to the steps array